src/app/modules/time-entries/pages/time-entries.component.spec.ts

```diff
@@ -427,4 +426,18 @@ describe('TimeEntriesComponent', () => {
       expect(entryData.customer_name).toContain('ioet Inc.');
     });
   }));
+
+  it('Should the entry be null if the flag is true', () => {
+    component.wasEditingExistingTimeEntry = true;
+    component.newEntry();
+    expect(component.entry).toBe(null);
+  });
+
+  it('Should the input persist the data if the flag is false', () => {
```

Comment:
`should persist the input data If I add a new entry and was not editing an existing time-entry.`

Reply:
I consider this test would be doing that if you enter a new entry, the data will persist as long as it is not edited.